api_v3/services/EntryServerNodeService.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Base class for entry server node
+ *
+ * @service entryServerNode
+ * @package api
+ * @subpackage services
+ */
+class EntryServerNodeService extends KalturaBaseService
+{
+
+	public function initService($serviceId, $serviceName, $actionName)
+	{
+		parent::initService($serviceId, $serviceName, $actionName);
+		$this->applyPartnerFilterForClass("entry");
+	}
+
+	/**
+	 * Adds a entry_user_node to the Kaltura DB.
+	 *
+	 * @action add
+	 * @param KalturaEntryServerNode $entryServerNode
+	 * @return KalturaEntryServerNode
+	 */
+	public function addAction(KalturaEntryServerNode $entryServerNode)
```

Comment:
If we want to move the logic from the backend to the actual media server we need this functionality available @kalturaguy